File: Projects/src/P01/Game/AI/Reaction.java
package P01.Game.AI;

public class Reaction implements Behavior {

    private Stimulus stimulus;
    private Action action;

    public Reaction(Stimulus stimulus, Action action) {
        this.stimulus = stimulus;
        this.action = action;
    }

    @Override
    public Action activate(Stimulus stimulus) {
        return stimulus == this.stimulus ? this.action : null;
    }
}

File: Projects/P03/src/lib/ecr/reaction.py
from lib.ecr.behaviour import Behaviour


class Reaction(Behaviour):

    def activate(self, perception):
        stimulus = self._detect_stimulus(perception)

        if stimulus is not None and stimulus is not False:
            response = self._generate_response(stimulus)
            return response

    def _generate_response(self, stimulus):
        "''"

    def _detect_stimulus(self, perception):
        "''"
